src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -2115,7 +2120,7 @@ class run_impl
       throw xrt_core::error(ENOSYS, "No auto-restart counters found for kernel");
 
     // TODO: find offset once in meta data
-    constexpr size_t counter_offset = 0x10;
+    constexpr size_t counter_offset = mailbox_auto_restart_cntr;
```

Comment:
We dont need this extra variable. Directly use mailbox_auto_restart_cntr in set_offset_value function